src/hotspot/share/classfile/verifier.cpp

```diff
@@ -255,7 +255,7 @@ bool Verifier::verify(InstanceKlass* klass, bool should_verify_class, TRAPS) {
         // or one of it's superclasses, we're in trouble and are going
         // to infinitely recurse when we try to initialize the exception.
         // So bail out here by throwing the preallocated VM error.
-        THROW_OOP_(Universe::virtual_machine_error_instance(), false);
+        THROW_OOP_(Universe::class_init_stack_overflow_error(), false);
```

Comment:
Should this be InternalError now?  That seems better than StackOverflow.

Reply:
Technically it's a stack overflow.  I don't think it's a reachable code path so it doesn't really matter.

Reply:
changed.